eng/_core/cmd/cmdscan/cmdscan.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Cmdscan runs the command given to it in os.Args and monitors the output for text patterns that
+// should be elevated to AzDO Pipeline warnings using AzDO logging commands. Timeline events can be
+// discovered more easily in the UI and by automated tools like runfo.
+//
+// If the pattern is associated with a known issue, the warning includes a link.
+//
+// Uses the "log issue" pipeline logging command: https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#logissue-log-an-error-or-warning
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+)
+
+var filters = []filter{
+	{
+		"access denied",
```

Comment:
I think this is fine for now, but feels a little odd to have them hard coded into source. As a future step, could we pass these in on the command line, then store them as a variable group in yaml that gets passed in by the CI? Feels like the actual set of issues belongs more with the infra than with code?

Reply:
Hmm, hadn't considered a variable group. Yeah, it would be nice to specify patterns in one place for all branches rather than per-branch. Having args (other than passing all args through to the inner command) makes the code a little more complex, but with a `--` separator, could avoid ambiguities, at least. I'll do that in a follow-up PR.